### ISSUE TITLE
GitHub API access_token query param is being deprecated

### ISSUE THREAD

**Issue by jakubgs** [octobox/octobox#2284]
🐞 **Problem**
Github is [deprecating `access_token` query parameter](https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param/) but OctoBox is still using it. I got this warning:
```
your application (Octobox) used an access token (with the User-Agent Faraday v0.17.3) as part of a query parameter to access an endpoint through the GitHub API.
```

🎯 **Goal**
The [necessary changes](https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param/#changes-to-make) need to be made to avoid OctoBox breaking on the November 13, 2020.

💡 **Possible solutions**
Follow the see [recommended way](https://developer.github.com/changes/2020-02-10-deprecating-auth-through-query-param/#changes-to-make) of using the API token.

📋  **Steps to solve the problem**
The API token should be passed to the API via an HTTP header instead of query parameter:
```
curl -H 'Authorization: token my_access_token' https://api.github.com/user/repos
```

**Comment by andrew** [comment]
@jakubgs can you provide a bit more log information to help me track down where the call is happening, things like which url you loaded and if there's any more stack trace? Also can you confirm you're running the latest version of octobox?

**Comment by jakubgs** [comment]
There was not stack trace because there was no error. It's just a warning email from GitHub.

As they state:

>During a brownout, authentication using query parameters will temporarily fail. The goal is to trigger alerts (assuming there are any) on our customers' services to help them find unmigrated authentication calls.
>
>The brownouts are scheduled for:
>
>* September 30, 2020
>  * From 7:00 AM UTC - 10:00 AM UTC
>  * From 4:00 PM UTC - 7:00 PM UTC
>* October 28, 2020
>  * From 7:00 AM UTC - 10:00 AM UTC
>  * From 4:00 PM UTC - 7:00 PM UTC

>Also can you confirm you're running the latest version of octobox?
```
 $ docker exec -it octobox sh
/usr/src/app # git rev-parse HEAD
82184acccd96848958e724142d2d79175524404e
```
It appears to be 82184acccd96848958e724142d2d79175524404e which appears to be from Jan 4.
I'm using the `latest` tag from `octoboxio/octobox` image:
```
 $ docker inspect octoboxio/octobox | jq -r '.[].Id'
sha256:be2b35b57ae497235f5eec868d749237e19e76635f1c9487baa1cc83caff61a5
 $ docker inspect octoboxio/octobox | jq -r '.[].Created'
2020-01-04T17:58:11.997361234Z
```
I see the `latest` tag has been updates:
https://hub.docker.com/layers/octoboxio/octobox/latest/images/sha256-0ac5fde93699c36307ff2ef4eabcd246bf8668bbdfbaafe934663abebdd958e9?context=explore
I'll deploy it.

**Comment by jakubgs** [comment]
I see that in 82184acccd96848958e724142d2d79175524404e you were using `4.15.0`:
https://github.com/octobox/octobox/blob/82184acccd96848958e724142d2d79175524404e/Gemfile.lock#L201
And now you're using `4.17.0`:
https://github.com/octobox/octobox/blob/873dfea1c34166bc636b3e5c907cec14871a7427/Gemfile.lock#L198
So it's likely that this has already been fixed by this upgrade.

**Comment by jakubgs** [comment]
Looks like this was indeed fixed in [`4.16.0`](https://github.com/octokit/octokit.rb/releases/tag/v4.16.0) with https://github.com/octokit/octokit.rb/pull/1192.

So I think this is already resolved.

**Comment by andrew** [comment]
Great, thanks for investing!
